input/script_utils.go

```diff
@@ -1200,14 +1312,21 @@ func receiverHtlcTapScriptTree(senderHtlcKey, receiverHtlcKey,
 // the tap leaf are returned.
 func ReceiverHTLCScriptTaproot(cltvExpiry uint32,
 	senderHtlcKey, receiverHtlcKey, revocationKey *btcec.PublicKey,
-	payHash []byte) (*HtlcScriptTree, error) {
+	payHash []byte, ourCommit bool) (*HtlcScriptTree, error) {
+
+	var hType htlcType
+	if ourCommit {
+		hType = htlcLocalIncoming
+	} else {
+		hType = htlcRemoteOutgoing
```

Comment:
same here, should this be `htlcRemoteIncoming`? Actually I'm a bit lost here...I remember someone told me the `incoming/outgoing` terminology always stands from our point of view, not theirs. So when we say `an incoming htlc on remote commit`, it means our incoming htlc on their commit, and to them it's an outgoing htlc from their PoV. Is this correct?

Reply:
See the above commit, I had it flipped as you mention when writing it originally, but after running the tests I realized that the sender/receiver here is sort of absolute. Which is why `localCommit` needed to be threaded thru in order to ensure we're signing the correct script.

Reply:
>  So when we say an incoming htlc on remote commit, it means our incoming htlc on their commit, and to them it's an outgoing htlc from their PoV. Is this correct?

That's correct. In this case we're the receiver of the HTLC, so it's incoming on our local commitment, but outgoing on their commitment. 

Check out the existing `genHTLCScript` method: 
https://github.com/lightningnetwork/lnd/blob/d930dcec40a2732b676a1b8b86571fb1943aa74e/lnwallet/commitment.go#L887-L903

Notice how we use `SenderHTLCScript` twice: 
  1. For an outgoing HTLC on our commitment. 
  2. For an incoming HTLC on their commitment. 

These are mirrors of the same state delta in a sense.